### PR TITLE
Add a deprecate_method function.

### PR DESCRIPTION
**Pull request by dagewa**
This makes it easier to emit a `DeprecationWarning` for methods of Boost Python classes. From an idea by @Anthchirp.

### CODE REVIEW

File: boost_adaptbx/tests/tst_deprecation_tools.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+def exercise():
+  with warnings.catch_warnings(record=True) as w:
+    warnings.simplefilter("always")
+
+    # Choose a Boost Python class for which to deprecate a method. Here we
+    # use rational.int.
+    from boost_adaptbx.boost import rational
+
+    original_value = rational.int().numerator()
+    deprecate_method(rational.int, "numerator")
```
Comment (Anthchirp) [inline boost_adaptbx/tests/tst_deprecation_tools.py:15]:
```suggestion
  # Choose a Boost Python class for which to deprecate a method. Here we
  # use rational.int.
  from boost_adaptbx.boost import rational

  original_value = rational.int().numerator()
  deprecate_method(rational.int, "numerator")
    
  with warnings.catch_warnings(record=True) as w:
    warnings.simplefilter("always")
```

to ensure the warning is thrown at call-time, not at setup-time.

**Review by Anthchirp — APPROVED** [review]
I would suggest putting the function into `boost_adaptbx/boost/python.py`, basically next to the `inject()` function.

File: boost_adaptbx/tests/tst_deprecation_tools.py (hunk 1 above)
Comment (dagewa) [reply boost_adaptbx/tests/tst_deprecation_tools.py:15]: Ok, can do. I was wrapping the first call to numerator too in order to be sure that didn't already raise a warning, but I suppose that isn't really necessary.

**Comment by dagewa** [comment]
Thanks! It looks like the CI is happy, so I'll merge.

**Pull request merged**